modules/collect/src/main/java/com/opengamma/collect/result/Result.java

```diff
@@ -466,6 +471,20 @@ public T getValue() {
   }
 
   /**
+   * Returns the actual result value if calculated successfully, or the specified
+   * default value if a failure occurred.
+   * <p>
+   * If this result is a success then the result value is returned.
+   * If this result is a failure then the default value is returned.
+   *
+   * @param defaultValue  the default value to return if the result is a failure
+   * @return either the result value or the default value
+   */
+  public T getValueOrElse(T defaultValue) {
+    return (isSuccess() ? value : defaultValue);
```

Comment:
Should ``defaultValue`` be nullable? Null isn't allowed as the value of a result so it would seem consistent to forbid it here too.